### PR TITLE
Refactor ImageDeleteModal to a functional component

### PR DESCRIPTION
**Pull request by jelly**
This now also disables the "select all" button if all checkboxes are checked.

### CODE REVIEW

File: src/ImageDeleteModal.jsx [hunk 1]
```diff
@@ -17,115 +17,98 @@ function sortTags(a, b) {
...
-        });
-    }
-
-    handleRemoveImage(tags, all) {
-        const Dialogs = this.context;
-        const image = this.props.imageWillDelete;
-
+const ImageDeleteModal = ({ imageWillDelete, onAddNotification }) => {
+    const Dialogs = useDialogs();
+    const repoTags = imageWillDelete.RepoTags ? imageWillDelete.RepoTags : [];
```
Comment (cockpituous) [inline src/ImageDeleteModal.jsx:22]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20221130-134924-205df97d-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageDeleteModal.jsx [hunk 2]
```diff
@@ -17,115 +17,98 @@ function sortTags(a, b) {
...
-        );
-    }
-}
+    };
+
+    return (
+        <Modal isOpen
+                 position="top" variant="medium"
+                 onClose={Dialogs.close}
+                 title={cockpit.format(_("Delete $0"), repoTags ? repoTags[0] : "")}
```
Comment (cockpituous) [inline src/ImageDeleteModal.jsx:85]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20221130-134924-205df97d-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageDeleteModal.jsx [hunk 3]
```diff
@@ -17,115 +17,98 @@ function sortTags(a, b) {
...
+                        );
+                    })}
+                </StackItem>
+            </Stack>
+            { repoTags.length > 2 && <Button isDisabled={repoTags.length === checkedTags.length} variant="link" onClick={pickAll}>{_("select all")}</Button> }
+        </Modal>
+    );
+};
+
+export default ImageDeleteModal;
```
Comment (KKoukiou) [inline src/ImageDeleteModal.jsx:114]: I actually love named exports. Why did you change it? It's also unrelated change.

Comment (jelly) [reply src/ImageDeleteModal.jsx]: Sure, I reverted it.

File: src/ImageDeleteModal.jsx [hunk 4]
```diff
@@ -17,115 +17,96 @@ function sortTags(a, b) {
...
+            client.untagImage(imageWillDelete.isSystem, imageWillDelete.Id, tag.substring(0, i), tag.substring(i + 1, tag.length))
                     .then(() => {
                         if (tags.length > 0)
-                            this.handleRemoveImage(tags, all);
+                            handleRemoveImage(tags, all);
                     })
                     .catch(ex => {
                         const error = cockpit.format(_("Failed to remove image $0"), tag);
-                        this.props.onAddNotification({ type: 'danger', error, errorDetail: ex.message });
+                        onAddNotification({ type: 'danger', error, errorDetail: ex.message });
```
Comment (cockpituous) [inline src/ImageDeleteModal.jsx:76]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20221130-163743-c39fb479-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageDeleteModal.jsx [hunk 5]
```diff
@@ -17,115 +17,96 @@ function sortTags(a, b) {
...
-        );
-    }
-}
+    };
+
+    return (
+        <Modal isOpen
+                 position="top" variant="medium"
+                 onClose={Dialogs.close}
+                 title={cockpit.format(_("Delete $0"), repoTags ? repoTags[0] : "")}
```
Comment (cockpituous) [inline src/ImageDeleteModal.jsx:85]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20221130-163743-c39fb479-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by jelly** [comment]
I can't reproduce this rawhide failure locally with the same packages which is super weird.

Both fail with TestApplication.testCheckpointRestoreCGroupsV2 which crun should fix:

```
[root@fedora-rawhide-127-0-0-2-2201 ~]# rpm -q runc crun podman criu kernel-core selinux-policy cockpit-podman cockpit-bridge
package runc is not installed
crun-1.7.2-1.fc38.x86_64
podman-4.3.1-1.fc38.x86_64
criu-3.17.1-3.fc37.x86_64
kernel-core-6.1.0-0.rc6.46.fc38.x86_64
kernel-core-6.1.0-0.rc7.51.fc38.x86_64
selinux-policy-38.1-1.fc38.noarch
cockpit-podman-57.11.gc39fb479-1.fc38.noarch
cockpit-bridge-280.1-1.fc38.x86_64
[root@fedora-rawhide-127-0-0-2-2201 ~]#
```

**Review by KKoukiou — APPROVED** [review]
The packit falures are agreeably unrelated. Let's fix them seperately
